app/models/participant.rb

```diff
@@ -124,7 +124,12 @@ def scores(questions)
       end
     end
 
-    scores[:total_score] = assignment.compute_total_score(scores)
+    # E1731 changes: Calling either LocalDbCalc or OnTheFlyCalc based on assignment status
+    scores[:total_score] = if self.assignment.local_scores_stored?
+                             LocalDbCalc.compute_total_score(assignment)
```

Comment:
As discussed in Expertiza meeting May 9th, this should be done by metaprogramming.